src/sentry/models/repository.py

```diff
@@ -2,22 +2,52 @@
 
 from django.db import models
 from django.utils import timezone
+from jsonfield import JSONField
 
+from sentry.constants import ObjectStatus
 from sentry.db.models import (
     BoundedPositiveIntegerField, Model, sane_repr
 )
+from sentry.signals import pending_delete
 
 
 class Repository(Model):
     __core__ = True
 
     organization_id = BoundedPositiveIntegerField(db_index=True)
     name = models.CharField(max_length=200)
+    url = models.URLField(null=True)
+    provider = models.CharField(max_length=64, null=True)
+    external_id = models.CharField(max_length=64, null=True)
```

Comment:
Should we disallow an `NULL` value for `external_id`? Otherwise, the `UNIQUE` constraint, from what I can tell, won't work very well. Since we'd be uniquing on `(org_id, provider, external_id)`, if external_id is ever `NULL`, it'll never violate the constraint since NULLs aren't considered equal. Unclear if this is desired or intended.